config-svc/Dockerfile

```diff
@@ -8,6 +8,13 @@ COPY . /app/
 
 RUN CGO_ENABLED=0 GOOS=linux go build -o /cmoscfg ./cmd/cmoscfg
 
+# We require various GCC stuff for tests
+FROM golang:1.17 AS unit-test
+COPY . /app/
+WORKDIR /app
+RUN go mod download && \
+    go test -trimpath -timeout 120s -v ./...
```

Comment:
Should this be an ENTRYPOINT rather than a RUN? If I'm understanding this right this'll run the tests at build time, is that intentional or am I missing something?

Reply:
Simplifies things really, we do want to do this during a build phase - otherwise you have to first build then run the container. It means you can just build and pass/fail in one command, you don't have to tag it, etc.